backend/clubs/views.py

```diff
@@ -506,6 +517,37 @@ def get_serializer_class(self):
             return WritableClubFairSerializer
         return ClubFairSerializer
 
+    @action(detail=True, methods=["get"])
+    def get_events_user_info(self):
+        """
+        Returns all events, grouped by id, with the number of participants currently in the meeting,
+        the officers or owners in the meeting, and the number of participants
+        who have already attended the meeting
+        """
+        fair = self.get_object()
+        clubs = fair.participating_clubs.all()
+        events = Event.objects.filter(
+            club__in=clubs,
+            type=Event.FAIR,
+            start_time_gte=fair.start_time,
+            end_time_lte=fair.end_time,
+        )
+
+        response = [
+            {
+                event.id: {
+                    "num_participants": len(
+                        ZoomMeetingVisit.objects.filter(event=event, leave_time=None)
+                    ),
+                    "officers": get_officers(event),
+                    "already_attended": len(ZoomMeetingVisit.objects.filter(event=event)),
```

Comment:
These two queries are too inefficient, as this will result in 1,200 SQL queries per request (given a ~600 event fair).

Reply:
Also this double counts users, as a user could join three times and that would count as 3 separate entries.